### PR TITLE
Remove less useful stuff

### PR DESCRIPTION
**Pull request by keymone**
Doing mostly tests  and waiting for packages to build i decided to spend some time and remove some things i believe are unnecessary complications. It's probably  better  to look at individual commits rather than PR as a whole.

The list includes but isn't limited to:

- FrozenDict, because dict is good enough (https://github.com/Yelp/Tron/commit/770fddf3c39f5be7afe9a1f64a7e6d06fce6257c)
- Enum in  favor of stdlib enum (https://github.com/Yelp/Tron/commit/7de24e93c003155c9dc1532146ef47267ee13092)
- FlockFile in favor of tron.utils.flock context function (
https://github.com/Yelp/Tron/commit/114ea29c855b86b9d59b174e9d9e52a01e34fc37)
- NoDaemonContext in favor of no_daemon_context context function (https://github.com/Yelp/Tron/commit/114ea29c855b86b9d59b174e9d9e52a01e34fc37)
- iteration.py because it doesn't really do much and that build_filtered_func is a bit confusing (https://github.com/Yelp/Tron/commit/bb1088bed236ddbaec6f204643e46fe3e35453ee)
- utils/tool_utils.py because it only has one function used in few old scripts and we should probably delete them too (https://github.com/Yelp/Tron/commit/ad9ea5d573ae42a51840028e7efb3455e18b4445)
- made trontimespec less defensive about absence of pytz (https://github.com/Yelp/Tron/commit/2f20338da7ae8e84c1e69c5b143fcb4e5fe2e5b4)
- remove dependency on six (https://github.com/Yelp/Tron/pull/584/commits/7ef597de8143fa4b0de94e263524a33c8f3f7c4e)

hopefully i'll also find time to rid us of MappingCollection and maybe CollectionProxy or replace it with  simpler implementation.

### CODE REVIEW

File: tools/migration/migrate_state.py [hunk 1]
```diff
@@ -73,8 +69,12 @@ def get_state_manager_from_config(config_path, working_dir):
     config_manager = manager.ConfigManager(config_path)
     config_container = config_manager.load()
     state_config = config_container.get_master().state_persistence
-    with tool_utils.working_dir(working_dir):
+    cwd = os.getcwd()
```
Comment (solarkennedy) [inline tools/migration/migrate_state.py]: I kinda liked this context manager.
do you think the try/finally is more technically correct?

Comment (keymone) [reply tools/migration/migrate_state.py]: yeah  i can keep it. i just didnt like having a whole separate file for it. i'll move directly to utils/__init__.py

File: tron/utils/__init__.py [hunk 2]
```diff
@@ -19,3 +27,47 @@ def next_or_none(iterable):
...
+
+@contextlib.contextmanager
+def flock(fd):
+    close = False
+    if isinstance(fd, str):
+        fd = open(fd, 'a')
+        close = True
+
+    try:
+        fcntl.lockf(fd, fcntl.LOCK_EX)
```
Comment (kawaiwanyelp) [inline tron/utils/__init__.py]: I'm pretty sure you want fcntl.LOCK_EX | fcntl.LOCK_NB' here. With just fcntl.LOCK_EX`, if someone else has locked the file, the process will block until that "someone else" releases the file.

**Comment by solarkennedy** [comment]
A lot of good stuff in here, but this is a large PR. Please make smaller changes next time. I know there are individual commits, but it is just easier to wrap your head around smaller changes.

My main concerns are the change in frozendict stuff, the flock changes, but otherwise I think this is great!

**Comment by solarkennedy** [comment]
@keymone can you revive this?

**Comment by keymone** [comment]
@solarkennedy thanks for reminding me :)

File: tron/utils/__init__.py (hunk 2 above)
Comment (keymone) [reply tron/utils/__init__.py]: fixed

**Pull request merged**